src/client/datascience/errors/errorHandler.ts

```diff
@@ -219,15 +227,78 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
+        message: string,
+        moreInfoLink: string | undefined,
+        cellToDisplayErrors?: NotebookCell
+    ) {
+        if (!message.includes(Commands.ViewJupyterOutput)) {
+            message = `${message} \n${DataScience.viewJupyterLogForFurtherInfo()}`;
+        }
+        void this.displayErrorsInCell(message, cellToDisplayErrors, moreInfoLink);
+        const buttons = moreInfoLink ? [Common.learnMore()] : [];
+        await this.applicationShell.showErrorMessage(message, ...buttons).then((selection) => {
+            if (selection === Common.learnMore() && moreInfoLink) {
                 this.browser.launch(moreInfoLink);
             }
         });
     }
+    private async handleJupyterStartupError(
+        failureInfo: KernelFailure | undefined,
+        error: JupyterConnectError,
+        kernelConnection: KernelConnectionMetadata,
+        cellToDisplayErrors?: NotebookCell
+    ) {
+        const failureInfoFromMessage =
+            failureInfo ||
+            analyzeKernelErrors(
+                error.message,
+                this.workspace.workspaceFolders,
+                kernelConnection.interpreter?.sysPrefix,
+                true
+            );
+        // Extract the python error message so we can display that.
+        let pythonError: string | undefined =
+            failureInfoFromMessage?.reason === KernelFailureReason.jupyterStartFailure ||
+            failureInfoFromMessage?.reason === KernelFailureReason.jupyterStartFailureOutdatedTraitlets
+                ? failureInfoFromMessage?.errorMessage
+                : undefined;
+        if (!pythonError) {
+            // Some times the error message is either in the message or the stderr.
+            pythonError = error.message
+                .splitLines({ removeEmptyEntries: true, trim: true })
+                .reverse()
+                .find((item) => item.toLowerCase().includes('Error: '));
+            pythonError =
+                pythonError ||
+                (error.stdErr || '')
+                    .splitLines({ removeEmptyEntries: true, trim: true })
+                    .reverse()
+                    .find((item) => item.toLowerCase().includes('Error: '));
+        }
+        const jupyterInterpreter = await this.jupyterInterpreter.getSelectedInterpreter();
+        const envDisplayName = jupyterInterpreter
+            ? `${jupyterInterpreter.displayName} (${getDisplayPath(
+                  jupyterInterpreter.path,
+                  this.workspace.workspaceFolders || []
+              )})`
+            : '';
+        if (
+            jupyterInterpreter &&
+            failureInfoFromMessage?.reason === KernelFailureReason.jupyterStartFailureOutdatedTraitlets
+        ) {
+            void this.showMessageWithMoreInfo(
+                DataScience.failedToStartJupyterDueToOutdatedTraitlets().format(envDisplayName, pythonError || ''),
+                'https://aka.ms/kernelFailuresJupyterTrailtletsOutdated',
```

Comment:
Looks like a misspell here, or is the aka link also misspelled?

Reply:
Yuck, i'll leave that spelling, as the link works and from what I know we can't delete those links.
Also no harm in that.